Fix MapView.toString when overriden

ashawley opened the pull request:
Since `MapView` isn't marked final and can be subclassed, the deprecated `stringPrefix` field needs to be set instead of `className`, for now, until the former is removed.  This is consistent with the rest of collections.

NthPortal commented on test/junit/scala/runtime/ScalaRunTimeTest.scala:
```diff
@@ -27,6 +27,8 @@ class ScalaRunTimeTest {
         stringOf(Array(Array("", 1, Array(5)), Array(1))))
 
+    assertEquals("""ArraySeq()""", stringOf(collection.mutable.ArraySeq.empty[AnyRef]))
+
```
why is this in `ScalaRunTimeTest`?

ashawley replied:
I added it to test something, but don't recall what.  I'll remove.

NthPortal replied:
the `MapView` one I don't think should be here either. I think there's already a `MapViewTest` maybe?

ashawley replied:
Good to know.  I've worked on this in fits and starts, so might have not existed when I began.

NthPortal commented:
`MapView` did not previously exist, so there are no prior collections extending it which override `stringPrefix` for which we need to maintain compatibility. I'd like to hear others' thoughts on this, specifically those who decided to change the method name (@julienrf or @szeiger?)

I think it would make more sense to `override final def stringPrefix = className` (although that's painfully toeing the line with asking for an infinite loop).

Has this come up in the wild?

ashawley commented:
> `MapView` did not previously exist

Indeed, that's true.  I'm only doing for consistency's sake.  The other problem is that overriding `stringPrefix`.  No, it's not something that's come up in the wild. 

> I think it would make more sense to `override final def stringPrefix = className`

Right, or to be able to make `stringPrefix` inaccessible in `MapView` so client code trying to override `stringPrefix` won't compile.

ashawley committed:
Fix MapView.toString when overriden

Since MapView isn't marked final and can be subclassed, the deprecated
stringPrefix field needs to be set instead of className, for now,
until it is removed.
